source/builtin_macros/src/syntax.rs

```diff
@@ -315,23 +315,62 @@ impl Visitor {
                 ));
             }
         }
-        if let Some(Ensures { token, mut exprs }) = ensures {
+        if let Some(Ensures { attrs, token, mut exprs }) = ensures {
             if exprs.exprs.len() > 0 {
                 for expr in exprs.exprs.iter_mut() {
                     self.visit_expr_mut(expr);
                 }
-                if let Some((p, ty)) = ret_pat {
-                    stmts.push(Stmt::Semi(
-                        Expr::Verbatim(
-                            quote_spanned!(token.span => ::builtin::ensures(|#p: #ty| [#exprs])),
-                        ),
-                        Semi { spans: [token.span] },
-                    ));
-                } else {
-                    stmts.push(Stmt::Semi(
-                        Expr::Verbatim(quote_spanned!(token.span => ::builtin::ensures([#exprs]))),
-                        Semi { spans: [token.span] },
-                    ));
+                let cont = match self.extract_quant_triggers(attrs, token.span) {
+                    Ok(
+                        found @ (ExtractQuantTriggersFound::Auto
+                        | ExtractQuantTriggersFound::Triggers(..)),
+                    ) => {
+                        if exprs.exprs.len() != 1 {
+                            let err = "when using #![trigger f(x)], only one ensures is allowed";
```

Comment:
Suggestion:

```
                        if exprs.exprs.len() == 0 {
                            let err = "when using #![trigger f(x)], at least one ensures is required";
```

This works fine:

```
            #[verifier::external_body]
            #[verifier::broadcast_forall]
            pub proof fn ab(a: A)
                ensures #![trigger a.f1()]
                    a.f3(),
                    a.f1() ==> a.f2(),
            {
            }
```

Reply:
This will produce 
```
::builtin::ensures([
  ::builtin::with_triggers( ( (a.f1(),), ), a.f3() ),
  a.f1() ==> a.f2(),
])
```
The two expressions are conjoined (&&) when we generate the AIR. It sounds like we interpret it correctly anyway, but it felt like it may subtly break our implicit assumptions on what goes in `with_triggers`, which may result in (silent) breakage later.

Reply:
We already handle this for the requires anyway.  When you write:

```
            pub proof fn ab(a: A)
                requires
                    a.f1(),
                ensures #![trigger a.f1()]
                    a.f2() && a.f3(),
```

poly.rs turns this into `a.f1() ==> with_triggers[a.f1()](a.f2() && a.f3())`.  The processing in triggers.rs is designed to handle this.  Even if at some point we want to change how we process with_triggers internally, I don't think that this should be visible as a restriction on user code written in Verus.

Reply:
Ah, that makes sense, I missed that part of the code path. Thanks for the clarification.